### ISSUE TITLE
SqlType::getResult logic around null is frustrating

### ISSUE THREAD

**Issue by ragnese** [vincentlauvlwj/Ktorm#150]
I have a custom type for one of my entity tables. When the SQL column is null, I want the value of the entity to have a particular value of this custom type. So I wrote:

```
object MyCustomSqlType : SqlType<MyCustom>(Types.VARCHAR, typeName = "myCustom") {
    override fun doSetParameter(ps: PreparedStatement, index: Int, parameter: MyCustom) {
        when (parameter) {
            MyCustom.NONE -> ps.setNull(index, Types.VARCHAR)
            else -> ps.setString(index, parameter.name)
        }
    }

    override fun doGetResult(rs: ResultSet, index: Int): MyCustom? {
        return rs.getString(index).let(MyCustom.Companion::fromNullableString)
    }
}
```

The `doGetResult` gets called and it correctly returns a `MyCustom.NONE`. However, `SqlType::getResult` ignores my value! Here is the code (SqlType.kt, line 58):
```
    fun getResult(rs: ResultSet, index: Int): T? {
        val result = doGetResult(rs, index)
        return if (rs.wasNull()) null else result
    }
```

Why do you need that `if` statement? Shouldn't `doGetResult` return null if it's really null? This totally overrides my attempt to map null values to some other value of my custom type.

Thank you!

**Comment by vincentlauvlwj** [comment]
Consider functions like `ResultSet.getInt`, `ResultSet.getLong`, `ResultSet.getDouble`. Their return types are Java primitive types and their values cannot be null. 

These functions would return 0 if the SQL result is null, so we have to call `wasNull` to check that and return null to users, which ensures the value we return is exactly what it is in the database.

**Comment by vincentlauvlwj** [comment]
I will make the function `SqlType.getResult` open so that you can override the default behaviour in your implementation.

**Comment by ragnese** [comment]
That would be great! Thank you!

**Comment by ragnese** [comment]
And I do understand the behavior as it relates to the "old" Java ResultSet stuff and nulls. Thank you for taking the time to explain. (Also, what silly behavior to return 0 for null!)

**Comment by vincentlauvlwj** [comment]
Fixed in version 3.0.0
